recipes/coincurve/meta.yaml

```diff
@@ -0,0 +1,80 @@
+{% set name = "coincurve" %}
+{% set version = "18.0.0" %}
+{% set commit = "1dd0675add2da67ad9e6acfc47eed041066f2ea1" %}
+{% set sha256 = "124d212b1a663a02a8086d2094e58605d114698f6e562e1212e0ca27a326d84d" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/ofek/coincurve/archive/{{ commit }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - update_pyproject.toml.patch
+    - update_setup.py.patch
+    - update_build.py.patch
+    - update_MANIFEST.in.patch
+    - update_coincurve_deprecated.patch
```

Comment:
Please add a note explaining why these patches are implemented. Some of them seem to be because we want to build libsecp in a separate feedstock (which is fine), but there also seem to be some that change the coincurve API. Probably, the patches that change the coincurve API should be merged upstream.

Reply:
`coincurve` provides a binding to `libsecp256k1`. However, libsecp256k1 is not a pypi package, so the author bundles it into his pypi package. But, with the requirements of conda-forge, `libsecp256k1` is a conda package, so providing `coincurve` requires modifying (significantly?) the way the package is built. In doing so, a few monior modifications were made to simplify the future updates in `libsecp256k1`

   - setup.py: Major refactoring to remove the no longer needed `'ibsecp256k1` build
   - setup_support.py: No longer needed (consequence of above)
   - build.py: Provision to more readily integrate updates from `libsecp256k1`
   - MANIFEST.in: removal of no longer needed references
   - coincurve_deprecated: Replace usage of deprecated calls
   - pyproject.toml: Simplify setup/pyproject
   
From the learning of conda-forge recipes, it seems that this list of changes disqualifies this PR and the proposed alternative will be to create an upstream that reflect these changes?

Reply:
Patching the build system is not the problem. Downstream users will NOT notice that you moved the metadata definition from setup.py to pyproject.toml. What concerns me is "coincurve_deprecated: Replace usage of deprecated calls". This patch seems to be modifying the coincurve API, something that downstream users of this package will notice.

Reply:
Agreed. It seemed counter productive to not fix issues in the original package and update to the latest `libsecp256k1` , but this is indeed opposite to the charter of conda-forge.
A proposal could be to revert these API changes and provide a separate package that implements the desired upgrades?

Reply:
You should use the same version of libsecp256k1 that the upstream coincurve package uses. If you really want to use a newer version of libsecp256k1. Work with the maintainers of coincurve to accept your upgrade to the new libsecp256k1 version. Wait until coincurve publishes a new release with your changes before publishing them here.

If coincurve is abandoned, then you can create your own fork of coincurve, but you cannot publish your fork as `coincurve` on this channel; you would need to publish it as `momentorc-coincurve` or change the project name to something new.

Reply:
Understood. Closing PR until conformance to conda-forge